var/spack/repos/builtin/packages/cool/package.py

```diff
@@ -0,0 +1,54 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Cool(CMakePackage):
+    """COOL provides specific software components and tools for the handling of the time
+       variation and versioning of the experiment conditions data."""
+
+    homepage = "https://coral-cool.docs.cern.ch/"
+    git      = "https://gitlab.cern.ch/lcgcool/cool.git"
+
+    tags = ['hep']
+
+    version('3.3.5', tag='COOL_3_3_5')
+    version('3.3.4', tag='COOL_3_3_4')
+    version('3.3.3', tag='COOL_3_3_3')
+
+    patch('cool.patch', level=0)
+
+    variant('binary_tag', default='auto')
+
+    depends_on('coral')
+    depends_on('root')
+    depends_on('vdt')
+    depends_on('xz')
+    depends_on('qt@5:', when='platform=linux')
+    depends_on('boost+chrono+system+thread')
+    depends_on('python')
+
+    def determine_binary_tag(self):
+        # As far as I can tell from reading the source code, `binary_tag`
+        # can be almost arbitrary.  The only real difference it makes is
+        # disabling oracle dependency for non-x86 platforms.
```

Comment:
Reading this comment, I wonder if it would be a better interface to users a variant called `oracle` or similar. Otherwise how can we convey this behavior to users?

Reply:
Setting the BINARY_TAG is required anyway. In the new version (3_3_7), this variable is also used to not add compiler to runtime path for ICC and MacOS builds. Also, these two packages (COOL and CORAL) are highly specialized and not supposed to be configurable.